src/lib/validator.ts

```diff
@@ -15,6 +15,32 @@ export const hasValue = (val: undefined | null | string): val is string => {
   return val !== undefined && val !== null && val !== "";
 };
 
+/**
+ * Returns true of val is not undefined, not null and not an empty string
+ * and can be converted to integer.
+ *
+ * @param val Value to inspect
+ */
+export const isInteger = (val: undefined | null | string): val is string => {
+  if (val === undefined || val === null || val === "") {
+    return false;
+  }
+  return /^[1-9]\d+$/.test(val);
+};
+
+/**
+ * Returns true of val is a port number.
+ *
+ * @param val Value to inspect
+ */
+export const isPortNumber = (val: undefined | null | string): val is string => {
```

Comment:
loosen the type constraint on `val` to be `unknown`

Reply:
Name should be more representative of functionality. Change to something like `isPortNumberString`

Reply:
I have no problem changing the function name if you really think that it is necessary. Please let me know

Reply:
> loosen the type constraint on `val` to be `unknown`

or we can remove the type guard :-)

Reply:
- rename the function.
- no reason to not have a typeguard

Reply:
accept